### PR TITLE
feat: Pull event/property definitions into its own service, so we can get it out of the plugin server

### PR DESCRIPTION
**Pull request by oliverb123**
This is the first step - much work needs to be done on the infrastructure around this, and then also on load testing and tuning.

Also I need to write tests.

oh yeah I also need to setup the sqlx offline stuff

### CODE REVIEW

**Review by bretthoerner — COMMENTED** [review]
I'm not deeply familiar with the existing properly definitions code, but this looks like a great start.

Also looks like we'll need to make a DB available (with the proper tables) even during the compilation step so that `sqlx::query!` works in CI.

File: rust/autocomplete/src/types.rs [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+pub struct PropertyDefinition {
+    pub id: Uuid,
+    pub team_id: TeamId,
+    pub name: String,
+    pub is_numerical: bool,
+    pub property_type: Option<PropertyValueType>,
+    #[serde(rename = "type")]
+    pub event_type: Option<PropertyParentType>,
+    pub group_type_index: Option<i32>, // This is an i16 in the DB, but the groups table uses i16's, so we use that here and only downconvert on insert/read
```
[the remark this answers is not in the document]
Comment (oliverb123) [reply rust/autocomplete/src/types.rs]: Clarified what I meant

**Comment by oliverb123** [comment]
TODO - a teams "first event received" capture happens in the plugin servers property definition code right now, and we need to emit a posthog event when that happens for comms. Moving first event tracking here means I need to also fire an event from here into the head of the queue on first event (which complicates how I handle first_event stuff a little)

File: rust/autocomplete/src/config.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+impl From<&KafkaConfig> for ClientConfig {
+    fn from(config: &KafkaConfig) -> Self {
+        let mut client_config = ClientConfig::new();
+        client_config
+            .set("bootstrap.servers", &config.kafka_hosts)
+            .set("statistics.interval.ms", "10000")
+            .set("group.id", config.consumer_group.clone())
+            .set("enable.auto.offset.store", "false"); // We store on a per-message basis anyway right now, but this is for later if we decide to work in batches
```
Comment (bretthoerner) [inline rust/autocomplete/src/config.rs:57]: I'm not so sure about this line. The offset store is just the memory store of the last offsets seen. `auto.commit` is what commits offsets, and it's still on. But I'm pretty sure it needs the offset store?

Either way, I don't see us committing offsets anywhere, so I think we want to use auto.commit and therefore want to remove this line and leave the offset store on?

(Probably not important for the code as it is now, but I'd rather not forget it for prod.)

**Review by bretthoerner — APPROVED** [review]

File: rust/autocomplete/src/config.rs (hunk 2 above)
Comment (oliverb123) [reply rust/autocomplete/src/config.rs]: You're right, if we fail to make forward progress we bail out of the loop and fall over, we shouldn't handle any offset committing manually at all (it's fine if falling over causes some messages to be re-processed)

**Pull request merged**